packages/react-dom/src/events/ReactBrowserEventEmitter.js

```diff
@@ -125,62 +151,74 @@ function getListeningForDocument(mountAt: any) {
-      switch (dependency) {
-        case TOP_SCROLL:
-          trapCapturedEvent(TOP_SCROLL, mountAt);
-          break;
-        case TOP_FOCUS:
-        case TOP_BLUR:
-          trapCapturedEvent(TOP_FOCUS, mountAt);
-          trapCapturedEvent(TOP_BLUR, mountAt);
-          // We set the flag for a single dependency later in this function,
-          // but this ensures we mark both as attached rather than just one.
-          isListening[TOP_BLUR] = true;
-          isListening[TOP_FOCUS] = true;
-          break;
-        case TOP_CANCEL:
-        case TOP_CLOSE:
-          if (isEventSupported(getRawEventName(dependency))) {
-            trapCapturedEvent(dependency, mountAt);
-          }
-          break;
-        case TOP_INVALID:
-        case TOP_SUBMIT:
-        case TOP_RESET:
-          // We listen to them on the target DOM elements.
-          // Some of them bubble so we don't want them to fire twice.
-          break;
-        default:
-          // By default, listen on the top level to all non-media events.
-          // Media events don't bubble so adding the listener wouldn't do anything.
-          const isMediaEvent = mediaEventTypes.indexOf(dependency) !== -1;
-          if (!isMediaEvent) {
-            trapBubbledEvent(dependency, mountAt);
-          }
-          break;
-      }
-      isListening[dependency] = true;
+    listenToDependency(dependency, listeningSet, mountAt, isLegacy);
+  }
+}
+
+function listenToDependency(
+  dependency: DOMTopLevelEventType,
+  listeningSet: Set<DOMTopLevelEventType>,
+  mountAt: Document | Element | Node,
+  isLegacy: boolean,
+): void {
+  if (!listeningSet.has(dependency)) {
+    switch (dependency) {
+      case TOP_SCROLL:
+        trapCapturedEvent(TOP_SCROLL, mountAt, isLegacy);
+        break;
+      case TOP_FOCUS:
+      case TOP_BLUR:
+        trapCapturedEvent(TOP_FOCUS, mountAt, isLegacy);
+        trapCapturedEvent(TOP_BLUR, mountAt, isLegacy);
+        // We set the flag for a single dependency later in this function,
+        // but this ensures we mark both as attached rather than just one.
+        listeningSet.add(TOP_BLUR);
+        listeningSet.add(TOP_FOCUS);
+        break;
+      case TOP_CANCEL:
+      case TOP_CLOSE:
+        if (isEventSupported(getRawEventName(dependency))) {
+          trapCapturedEvent(dependency, mountAt, isLegacy);
+        }
+        break;
+      case TOP_INVALID:
+      case TOP_SUBMIT:
+      case TOP_RESET:
+        // We listen to them on the target DOM elements.
+        // Some of them bubble so we don't want them to fire twice.
+        break;
+      default:
+        // By default, listen on the top level to all non-media events.
+        // Media events don't bubble so adding the listener wouldn't do anything.
+        const isMediaEvent = mediaEventTypes.indexOf(dependency) !== -1;
+        if (!isMediaEvent) {
+          trapBubbledEvent(dependency, mountAt, isLegacy);
+        }
```

Comment:
Back to the capture thing. If we could ensure execution order is maintained, we could just capture all the media events and get rid of this block.

That's out of the context of this PR, but still I wonder.